driver/modern_bpf/programs/tail_called/events/syscall_dispatched_events/mlock.bpf.c

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2022 The Falco Authors.
+ *
+ * This file is dual licensed under either the MIT or GPL 2. See MIT.txt
+ * or GPL2.txt for full copies of the license.
+ */
+
+#include <helpers/interfaces/fixed_size_event.h>
+
+/*=============================== ENTER EVENT ===========================*/
+
+SEC("tp_btf/sys_enter")
+int BPF_PROG(mlock_e,
+	     struct pt_regs *regs,
+	     long id)
+{
+	struct ringbuf_struct ringbuf;
+	if(!ringbuf__reserve_space(&ringbuf, MLOCK_E_SIZE))
+	{
+		return 0;
+	}
+
+	ringbuf__store_event_header(&ringbuf, PPME_SYSCALL_MLOCK_E);
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	ringbuf__submit_event(&ringbuf);
+
+	return 0;
+}
+
+/*=============================== ENTER EVENT ===========================*/
+
+/*=============================== EXIT EVENT ===========================*/
+
+SEC("tp_btf/sys_exit")
+int BPF_PROG(mlock_x,
+	     struct pt_regs *regs,
+	     long ret)
+{
+	struct ringbuf_struct ringbuf;
+	if(!ringbuf__reserve_space(&ringbuf, MLOCK_X_SIZE))
+	{
+		return 0;
+	}
+
+	ringbuf__store_event_header(&ringbuf, PPME_SYSCALL_MLOCK_X);
+
+	/*=============================== COLLECT PARAMETERS  ===========================*/
+
+	/* Parameter 1: res (type: PT_ERRNO)*/
+	ringbuf__store_s64(&ringbuf, ret);
+
+	/* Parameter 2: addr (type: PT_UINT64) */
+	unsigned long val = extract__syscall_argument(regs, 0);
```

Comment:
Thanks! Noted for future MRs =)

Reply:
Btw sorry I haven't been as responsive here lately, I should be able to get to all of the comments on these MRs later this week!

IIUC this one is good to go though?

Reply:
> Btw sorry I haven't been as responsive here lately, I should be able to get to all of the comments on these MRs later this week!

Don't worry at all!

> IIUC this one is good to go though?

Oh yes, it is! I cannot approve it because I've pushed on it, but it LGTM!